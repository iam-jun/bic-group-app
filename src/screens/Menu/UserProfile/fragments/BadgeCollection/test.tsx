import React from 'react';
import {
  View, Dimensions, FlatList, StyleSheet,
} from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import Avatar from '~/baseComponents/Avatar';
import Icon from '~/baseComponents/Icon';
import spacing, { borderRadius } from '~/theme/spacing';

const Grid = ({ data }: {data: any[]}) => {
  const deviceWidth = Dimensions.get('window').width;

  const itemWidth = 48; // Constant item width
  const numColumns = Math.floor((deviceWidth - 10 * 2) / (itemWidth + 6 * 2));
  const rowWidth = numColumns * (itemWidth + 6 * 2);
  const paddingLeft = (deviceWidth - (10 * 2) - rowWidth) / 2;

  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const styles = themeStyles(theme);

  const isSelected = false;

  const renderGridItem = ({ item, index }: any) =>
    // Render each grid item here
    (
      <View
        key={index}
        style={{
          margin: 6,
          borderWidth: isSelected ? 2 : 1,
          borderColor: isSelected ? colors.purple50 : colors.neutral5,
          borderRadius: borderRadius.pill,
        }}
      >
        <Avatar.Medium
          isRounded
          source={{ uri: item?.iconUrl }}
        />
        {isSelected
          ? (
            <View style={styles.iconChose}>
              <Icon size={10} icon="Check" tintColor={colors.white} />
            </View>
          )
          : null}
      </View>
    );

  return (
    <View style={{
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      paddingHorizontal: 10,
      paddingBottom: 10,
    }}
    >

      <View style={{
        // paddingLeft,
      }}
      >
        <FlatList
          data={data}
          renderItem={renderGridItem}
          keyExtractor={(item, index) => index.toString()}
          numColumns={numColumns}
        />
      </View>
    </View>
  );
};

const themeStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      flex: 1,
      paddingTop: spacing.padding.large,
    },
    iconChose: {
      backgroundColor: colors.purple50,
      borderRadius: borderRadius.pill,
      height: 20,
      width: 20,
      alignItems: 'center',
      justifyContent: 'center',
      position: 'absolute',
      zIndex: 1,
      right: 0,
    },
  });
};

export default Grid;
