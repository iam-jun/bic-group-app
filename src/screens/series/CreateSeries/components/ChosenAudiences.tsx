import React from 'react';
import { StyleSheet } from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import Text from '~/beinComponents/Text';
import { useBaseHook } from '~/hooks';
import spacing from '~/theme/spacing';
import ButtonWrapper from '~/baseComponents/Button/ButtonWrapper';

interface Props {
    audiences: any;
    disabled?: boolean;
    onPressAudiences: ()=>void;
}

const ChosenAudiences = ({ audiences, disabled, onPressAudiences }: Props) => {
  const { t } = useBaseHook();

  const theme: ExtendedTheme = useTheme();
  const styles = themeStyles(theme);

  return (
    <ButtonWrapper
      disabled={disabled}
      style={[styles.container, { opacity: disabled ? 0.5 : 1 }]}
      onPress={onPressAudiences}
      testID="create_post_chosen_audiences"
    >
      <Text.BodyM numberOfLines={2} color={theme.colors.neutral40} style={styles.textPostTo}>
        {`${t('post:post_to')} `}
        <Text.BodyMMedium color={theme.colors.neutral60}>{audiences?.count || 0}</Text.BodyMMedium>
        {': '}
        <Text.BodyMMedium color={theme.colors.neutral80} testID="create_post_chosen_audiences.names">{audiences?.names || ''}</Text.BodyMMedium>
      </Text.BodyM>
    </ButtonWrapper>
  );
};

const themeStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      flexDirection: 'row',
      paddingHorizontal: spacing?.padding.large,
      paddingVertical: spacing?.padding.small,
      alignItems: 'center',
      borderBottomColor: colors.neutral5,
      borderBottomWidth: 1,
    },
    textPostTo: {
    },
  });
};

export default ChosenAudiences;
