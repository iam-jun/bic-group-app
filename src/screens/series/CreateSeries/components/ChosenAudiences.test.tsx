import * as React from 'react';
import { fireEvent, renderWithRedux } from '~/test/testUtils';
import ChosenAudiences from './ChosenAudiences';

describe('ChosenAudiences component', () => {
  const audience = { count: 2, names: 'audience 1, audience 2' };

  it('render correctly', () => {
    const handlePressAudiences = jest.fn();
    const wrapper = renderWithRedux(<ChosenAudiences
      audiences={audience}
      onPressAudiences={handlePressAudiences}
    />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should call onPressAudiences when click component', () => {
    const handlePressAudiences = jest.fn();
    const wrapper = renderWithRedux(<ChosenAudiences
      audiences={audience}
      onPressAudiences={handlePressAudiences}
    />);

    const component = wrapper.getByTestId('create_post_chosen_audiences');
    expect(component).toBeDefined();

    fireEvent.press(component);
    expect(handlePressAudiences).toBeCalled();
  });

  it('render correctly when audiences is empty', () => {
    const handlePressAudiences = jest.fn();
    const wrapper = renderWithRedux(<ChosenAudiences
      audiences={{}}
      onPressAudiences={handlePressAudiences}
    />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should disabled component when prop disabled = true', () => {
    const handlePressAudiences = jest.fn();
    const wrapper = renderWithRedux(<ChosenAudiences
      audiences={audience}
      onPressAudiences={handlePressAudiences}
      disabled
    />);
    expect(wrapper.toJSON()).toMatchSnapshot();

    const component = wrapper.getByTestId('create_post_chosen_audiences');
    expect(component).toBeDefined();

    fireEvent.press(component);
    expect(handlePressAudiences).toBeCalledTimes(0);
  });
});
