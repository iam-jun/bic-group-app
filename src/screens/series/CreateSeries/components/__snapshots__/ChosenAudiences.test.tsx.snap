// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChosenAudiences component render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#E3E6F2",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
  testID="create_post_chosen_audiences"
>
  <Text
    numberOfLines={2}
    style={
      Array [
        Object {
          "color": "#212745",
          "fontFamily": "BeVietnamPro-Light",
          "fontSize": 14,
        },
        Object {
          "color": "#353E6E",
        },
        Object {},
      ]
    }
  >
    Post to 
    <Text
      style={
        Array [
          Object {
            "color": "#212745",
            "fontFamily": "BeVietnamPro-Medium",
            "fontSize": 15,
          },
          Object {
            "color": "#282E53",
          },
          undefined,
        ]
      }
    >
      2
    </Text>
    : 
    <Text
      style={
        Array [
          Object {
            "color": "#212745",
            "fontFamily": "BeVietnamPro-Medium",
            "fontSize": 15,
          },
          Object {
            "color": "#181C32",
          },
          undefined,
        ]
      }
      testID="create_post_chosen_audiences.names"
    >
      audience 1, audience 2
    </Text>
  </Text>
</View>
`;

exports[`ChosenAudiences component render correctly when audiences is empty 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#E3E6F2",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
  testID="create_post_chosen_audiences"
>
  <Text
    numberOfLines={2}
    style={
      Array [
        Object {
          "color": "#212745",
          "fontFamily": "BeVietnamPro-Light",
          "fontSize": 14,
        },
        Object {
          "color": "#353E6E",
        },
        Object {},
      ]
    }
  >
    Post to 
    <Text
      style={
        Array [
          Object {
            "color": "#212745",
            "fontFamily": "BeVietnamPro-Medium",
            "fontSize": 15,
          },
          Object {
            "color": "#282E53",
          },
          undefined,
        ]
      }
    >
      0
    </Text>
    : 
    <Text
      style={
        Array [
          Object {
            "color": "#212745",
            "fontFamily": "BeVietnamPro-Medium",
            "fontSize": 15,
          },
          Object {
            "color": "#181C32",
          },
          undefined,
        ]
      }
      testID="create_post_chosen_audiences.names"
    >
      
    </Text>
  </Text>
</View>
`;

exports[`ChosenAudiences component should disabled component when prop disabled = true 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#E3E6F2",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "opacity": 0.5,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
  testID="create_post_chosen_audiences"
>
  <Text
    numberOfLines={2}
    style={
      Array [
        Object {
          "color": "#212745",
          "fontFamily": "BeVietnamPro-Light",
          "fontSize": 14,
        },
        Object {
          "color": "#353E6E",
        },
        Object {},
      ]
    }
  >
    Post to 
    <Text
      style={
        Array [
          Object {
            "color": "#212745",
            "fontFamily": "BeVietnamPro-Medium",
            "fontSize": 15,
          },
          Object {
            "color": "#282E53",
          },
          undefined,
        ]
      }
    >
      2
    </Text>
    : 
    <Text
      style={
        Array [
          Object {
            "color": "#212745",
            "fontFamily": "BeVietnamPro-Medium",
            "fontSize": 15,
          },
          Object {
            "color": "#181C32",
          },
          undefined,
        ]
      }
      testID="create_post_chosen_audiences.names"
    >
      audience 1, audience 2
    </Text>
  </Text>
</View>
`;
